point_angle_range_to_polygon.py

```diff
@@ -12,47 +12,67 @@ def createPointAngleRangePolygons(points, outputPolygons, azimuthField, rangeFie
     azimuthField = azimuthField
     rangeField = rangeField
     beamWidthField = beamWidthField
-    #Polygon feature class will created by the tool. 
     outputPolygonFeatures = outputPolygons
     polygons = []
-    
-    cursorFields = ["SHAPE@", azimuthField, rangeField, beamWidthField]
+    #Setup field list for point search cursor and polygon insert cursor    
+    cursorFields = [f.name for f in arcpy.ListFields(pointFeatures)]
+    cursorFields.remove('SHAPE')
+    cursorFields.append("SHAPE@")
+    polygonInsCursor = arcpy.da.InsertCursor(outputPolygonFeatures, cursorFields)
     with arcpy.da.SearchCursor(pointFeatures, cursorFields) as cursor:
         for row in cursor:
-            centerPoint = row[0]
-            azimuth = row[1]
-            rangeDist = row[2]
-            beamWidth = row[3]
+            centerPoint = row[len(cursorFields) - 1]
+            azimuth = row[cursor.fields.index(azimuthField)]
+            rangeDist = row[cursor.fields.index(rangeField)]
+            beamWidth = row[cursor.fields.index(beamWidthField)]
             
             startAngle = (azimuth + (360 - (beamWidth / 2.0))) % 360
-            print startAngle
             radiusPoints = [centerPoint.centroid]
             #Create a point every 1 degree around the arc
             for i in range(int(beamWidth) + 1):
                 angle = (startAngle + i) % 360
-                print angle
                 radPoint = centerPoint.pointFromAngleAndDistance(angle, rangeDist)
                 radiusPoints.append(radPoint.centroid)
-                
-            polygons.append(
-                arcpy.Polygon(
-                    arcpy.Array(radiusPoints), 
-                    centerPoint.spatialReference))
-            print 
+            #Use point row as polygon insert row with new polygon geometery subsituted for point
+            newRow = [f for f in row]
```

Comment:
Cursor rows are tuples so I did have to duplicate it to replace the point geometry with the polygon.
Using the built-in list() makes more sense though so I changed it to `list(row) `.